Final Model.ipynb

```diff
@@ -1035,64 +836,251 @@
+       "      <td>UravuSolla-P.Susheela</td>\n",
+       "      <td>0</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                              file  \\\n",
+       "0  JealousKindOfFella-GarlandGreen   \n",
+       "1     InitialsB.B.-SergeGainsbourg   \n",
+       "2           MelodyTwist-LordMelody   \n",
+       "3            MiBombaSonó-CeliaCruz   \n",
+       "4            UravuSolla-P.Susheela   \n",
+       "\n",
+       "                                              lyrics  \n",
+       "0  3 ContributorsJealous Kind of Fella Lyrics[Int...  \n",
+       "1  14 ContributorsInitials B.B. Lyrics[Paroles de...  \n",
+       "2  1 ContributorJuice (haeY haeY) LyricsHook (Kro...  \n",
+       "3  2 ContributorsMi Bomba Sonó LyricsAcó, mi bomb...  \n",
+       "4                                                  0  "
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "lyrics.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
    "id": "f4a0865c",
    "metadata": {
     "scrolled": true
    },
    "outputs": [
     {
-     "name": "stdout",
+     "name": "stderr",
      "output_type": "stream",
      "text": [
-      "0\n",
-      "1\n",
-      "2\n",
-      "3\n",
-      "4\n",
-      "5\n",
-      "6\n",
-      "7\n",
-      "8\n",
-      "9\n"
+      "100%|██████████| 2947/2947 [00:09<00:00, 297.22it/s]\n"
      ]
     }
    ],
    "source": [
     "X_lyrics = []\n",
-    "for index, row in dataset.iterrows():\n",
-    "    print(index)\n",
-    "    track = row[\"track\"]\n",
-    "    track = track.replace(\" \", \"\")\n",
-    "    \n",
-    "    artist = row[\"artist\"]\n",
-    "    artist = artist.replace(\" \", \"\")\n",
-    "    \n",
-    "    title = track + \"-\" + artist\n",
-    "    \n",
-    "    lyrics_path = os.path.join(\"lyrics\", title + \".txt\")\n",
-    "    with open(lyrics_path, \"r\") as file:\n",
-    "        file_contents = file.read()\n",
-    "        X_lyrics.append(file_contents)\n",
+    "for file in tqdm(files_subset):  \n",
+    "    lyrics_string = lyrics.loc[lyrics['file'] == file, 'lyrics'].values[0]\n",
+    "    X_lyrics.append(lyrics_string)\n",
     "\n",
     "X_lyrics = np.array(X_lyrics)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 13,
    "id": "2af7d15f",
    "metadata": {},
-   "outputs": [],
-   "source": []
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "5d923933",
-   "metadata": {},
-   "outputs": [],
-   "source": []
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "2947"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "len(X_lyrics)"
+   ]
   },
   {
    "cell_type": "code",
```

Comment:
In the following cell (execution 36), why do you set num_genres = 3?

Reply:
same question, there appeared to be >3 genres in the raw dataset i was looking at?

Reply:
i think this was a prior version - before the midterm i believe